advanced_source/super_resolution_with_onnxruntime.py

```diff
@@ -0,0 +1,285 @@
+batch_size = 1    # just a random number
+
+# Initialize model with the pretrained weights
+map_location = lambda storage, loc: storage
+if torch.cuda.is_available():
+    map_location = None
+torch_model.load_state_dict(model_zoo.load_url(model_url, map_location=map_location))
+
+# set the train mode to false since we will only run the forward pass.
+torch_model.train(False)
+
+
+######################################################################
+# Exporting a model in PyTorch works via tracing or scripting. This
+# tutorial will use as an example a model exported by tracing. 
+# To export a model, you call the ``torch.onnx.export()`` function.
+# This will execute the model, recording a trace of what operators
+# are used to compute the outputs.
+# Because ``_export`` runs the model, we need to provide an input
+# tensor ``x``. The values in this can be random as long as it is the
+# right type and size.
+#
+# To learn more details about PyTorch's export interface, check out the
+# `torch.onnx documentation <https://pytorch.org/docs/master/onnx.html>`__.
+#
+
+# Input to the model
+x = torch.randn(batch_size, 1, 224, 224, requires_grad=True)
+
+# Export the model
+torch_out = torch.onnx._export(torch_model,               # model being run
+                               x,                         # model input (or a tuple for multiple inputs)
+                               "super_resolution.onnx",   # where to save the model (can be a file or file-like object)
+                               export_params=True,        # store the trained parameter weights inside the model file
+                               opset_version=10,          # the onnx version to export the model to
+                               do_constant_folding=True,  # wether to execute constant folding for optimization
+                               input_names = ['input'],   # the model's input names
+                               output_names = ['output'], # the model's output names
+                               dynamic_axes={'input' : {0 : 'batch_size'},    # variable lenght axes
+                                             'output' : {0 : 'batch_size'}})
+
+######################################################################
+# ``torch_out`` is the output after executing the model. Normally you can
+# ignore this output, but here we will use it to verify that the model we
+# exported computes the same values when run in onnxruntime.
+#
+# But before verifying the model's output with onnxruntime, we will check
+# the onnx model with onnx's API. This will verify the model's structure
+# and confirm that the model has a valid schema. 
+
+import onnx
+
+onnx_model = onnx.load("super_resolution.onnx")
+onnx.checker.check_model(onnx_model)
+
+
+######################################################################
+# Now let's create an onnxruntime session. This part can normally be
```

Comment:
Maybe we can expand on this a little, saying a few different things:
1) We are using ORT's Python APIs.
2) In ORT we need to create an inference session for the model with chosen configuration parameters (here we use the default config).
3) Once the session is created, we evaluate the model using the run() api.